File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.swift
class Solution {
    func maxAdjacentDistance(_ nums: [Int]) -> Int {
        var maxDistance = 0
        let n = nums.count

        for i in 0..<n {
            let next = (i + 1) % n // circular array
            let distance = abs(nums[i] - nums[next])
            maxDistance = max(maxDistance, distance)
        }

        return maxDistance
    }
}

File: 3747-maximum-difference-between-adjacent-elements-in-a-circular-array/3747-maximum-difference-between-adjacent-elements-in-a-circular-array.dart
class Solution {
  int maxAdjacentDistance(List<int> nums) {
    int max = 0;
    int n = nums.length;

    for (int i = 0; i < n; i++) {
      int next = (i + 1) % n; // circular array
      int sol = (nums[i] - nums[next]).abs();
      max = max > sol ? max : sol;
    }

    return max;
  }
}
